src/components/cc-logs/cc-logs.types.ts

```diff
@@ -0,0 +1,40 @@
+import { TemplateResult } from "lit-html";
+
+interface LogsStateLoading {
+  state: "loading";
+}
+
+interface LogsStateError {
+  state: "error";
+}
+
+interface LogsStateLoaded {
+  state: "loaded";
+  logs: Array<Log>;
+}
+
+type LogsState = LogsStateLoading | LogsStateError | LogsStateLoaded
+
+interface Log {
+  timestamp: number;
+  message: string;
+  metadata: Array<Metadata>
+}
+
+interface Metadata {
+  name: string;
+  value: string | number | boolean;
+}
+
+type MetadataRenderer = (metadata: Metadata, { index: number, log: Log }) => TemplateResult;
+
+type MessageRenderer = (message: string, { log: Log }) => TemplateResult;
+
+interface TimestampFormat {
+  pattern: 'datetime' | 'time';
+  precision: 'milliseconds' | 'seconds';
+  utc: boolean;
+  showZoneOffset: boolean;
+}
+
+type TimestampDisplay = 'hidden' | TimestampFormat;
```

Comment:
**suggestion:** we could use the same logic as the states here to provide a type that would always be an object:
```ts
interface TimestampHidden {
  visibility: 'hidden';
}

interface TimestampVisible {
  visibility: 'visible';
  pattern: 'datetime' | 'time';
  precision: 'milliseconds' | 'seconds';
  utc: boolean;
  showZoneOffset: boolean;
}

export type TimestamDisplay = TimestampHidden | TimestampVisible;
```

I suggest this because the way it currently works, it breaks the `attribute` option in the `timestampDisplay` prop. The attribute (and lit converter) expects an object but sometimes it will get a string and sometimes an object.

This would also allow us to do something like this in the `_renderLog ()` subrender:
```js
  ${this.timestampDisplay.visibility === 'visible'
      ? this._renderLogTimestamp(log, timestampFormatter)
      : ''}
```